source/s3_auto_ranged_put.c

```diff
@@ -174,6 +136,19 @@ static void s_s3_meta_request_auto_ranged_put_destroy(struct aws_s3_meta_request
     aws_mem_release(meta_request->allocator, auto_ranged_put);
 }
 
+static void s_s3_auto_ranged_put_finish(
+    struct aws_s3_meta_request *meta_request,
+    struct aws_s3_request *failed_request,
+    int error_code) {
+    struct aws_s3_auto_ranged_put *auto_ranged_put = meta_request->impl;
+    s_s3_auto_ranged_put_lock_synced_data(auto_ranged_put);
+    auto_ranged_put->synced_data.error_code = error_code;
+    auto_ranged_put->synced_data.failed_request = failed_request;
+    s_s3_auto_ranged_put_unlock_synced_data(auto_ranged_put);
+    /* state of meta request has been set now, and the state of auto ranged put will set properly by the task */
+    aws_s3_meta_request_push_to_client(meta_request);
```

Comment:
lost at all the task in client... But, what I see the reason for meta request not get cleaned up is this line will push another task to client, and require another refcount on meta request, but the client does not really process the meta request twice. Probably because of this condition. @rccarper 
https://github.com/awslabs/aws-c-s3/blob/main/source/s3_client.c#L1292-L1296